### PR TITLE
Some more op documentation

### PR DESCRIPTION
**Pull request by diyessi**

### CODE REVIEW

File: doc/sphinx/source/ops/get_output_element.rst [hunk 1]
```diff
@@ -0,0 +1,49 @@
+.. get_output_element.rst:
+
+################
+GetOutputElement
+################
+
+.. code-block:: cpp
+
+   GetOutputElement  // Operation to get an output from an op
```
Comment (aprocter) [inline doc/sphinx/source/ops/get_output_element.rst]: Maybe `select an single output from a multi-output op`

File: doc/sphinx/source/ops/max_pool.rst [hunk 2]
```diff
@@ -0,0 +1,83 @@
+.. max_pool.rst:
+
+#######
+MaxPool
+#######
+
+.. code-block:: cpp
+
+   MaxPool  // Common MaxPool operations
```
Comment (aprocter) [inline doc/sphinx/source/ops/max_pool.rst]: Not sure what "common" means here

File: doc/sphinx/source/ops/max_pool.rst [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
++-----------------+-------------------------+--------------------------------+
+
+Max pooling takes as its input a data batch tensor of shape
+:math:`(N,C,d_1,\dots,d_n)` where :math:`n > 0`, every :math:`d_i >
+0`, and where :math:`N` is the batch size, and :math:`C > 0` is the
+number of channels (sometimes called features).  The dimensions
+:math:`(d_1,\dots,d_n` correspond to the shape of an
+:math:`n`-dimensional data item in a batch. For example, where
+:math:`n=2`, the data may represent a two-dimensional image.  It also
+takes two parameters:
```
Comment (aprocter) [inline doc/sphinx/source/ops/max_pool.rst]: s/parameters/attributes/ (and maybe s/takes/has/)

File: doc/sphinx/source/ops/pad.rst [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+   d'_i = \mathtt{padding\_below}_i+d_i\cdot(\mathtt{padding\_interior}_i)+\mathtt{padding\_above}_i
+
+
+Takes an input tensor of shape :math:`(d_1,\dots,d_n)` and pads by inserting a scalar :math:`x` supplied as input, in three possible ways:
+
+1. *exterior padding* inserts copies of :math:`x` *below or above* the bounds of existing rows, columns, etc.,
+2. *interior padding* inserts copies of :math:`x` *between* rows, columns, etc., or
+3. both of the above.
+
+The number and position of elements to be inserted along a given axis is determined by three parameters:
```
Comment (aprocter) [inline doc/sphinx/source/ops/pad.rst]: parameters->attributes

File: doc/sphinx/source/ops/pad.rst [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+              42 42 42 42 42 42 42 42 42
+              42 42  1 42 42  2 42 42  3
+    1 2 3     42 42 42 42 42 42 42 42 42
+    4 5 6 --> 42 42  4 42 42  5 42 42  6
+    7 8 9     42 42 42 42 42 42 42 42 42
+              42 42  7 42 42  8 42 42  9
+              42 42 42 42 42 42 42 42 42
+
+In other words we have inserted one new row between each pair of adjacent rows, two new columns between each pair of adjacent columns, one new row at the top and two new columns on the left, and one new row at the bottom and zero new columns on the right; then filled the new rows and columns with 42.
+
```
Comment (indie) [inline doc/sphinx/source/ops/pad.rst]: This probably should be a `:note:` syntax in rst to call it out more explicitly, but something  can add later.

**Pull request merged**